src/routes/files/handlers.test.js

```diff
@@ -0,0 +1,137 @@
+import { validate } from 'uuid';
+import db, {
+  File,
+  ActivityReport,
+  User,
+  Permission,
+} from '../../models';
+import app from '../../app';
+import s3Uploader from '../../lib/s3Uploader';
+import SCOPES from '../../middleware/scopeConstants';
+
+const request = require('supertest');
+
+const ORIGINAL_ENV = process.env;
+
+jest.mock('../../lib/s3Uploader');
+
+const mockUser = {
+  id: 200,
+  homeRegionId: 1,
+  permissions: [
+    {
+      userId: 200,
+      regionId: 5,
+      scopeId: SCOPES.READ_WRITE_REPORTS,
+    },
+    {
+      userId: 200,
+      regionId: 6,
+      scopeId: SCOPES.READ_WRITE_REPORTS,
+    },
+  ],
+};
+
+const mockSession = jest.fn();
+mockSession.userId = mockUser.id;
+
+const reportObject = {
+  activityRecipientType: 'grantee',
+  status: 'draft',
+  userId: mockUser.id,
+  lastUpdatedById: mockUser.id,
+  resourcesUsed: 'test',
+};
+
+describe('File Upload Handlers happy path', () => {
+  let user;
+  let report;
+  let fileId;
+  beforeAll(async () => {
+    user = await User.create(mockUser, { include: [{ model: Permission, as: 'permissions' }] });
+    report = await ActivityReport.create(reportObject);
+    process.env.NODE_ENV = 'test';
+    process.env.BYPASS_AUTH = 'true';
+  });
+  afterAll(async () => {
+    await File.destroy({ where: {} });
+    await ActivityReport.destroy({ where: { } });
+    await User.destroy({ where: { id: user.id } });
+    process.env = ORIGINAL_ENV; // restore original env
+  });
+  beforeEach(() => {
+    s3Uploader.mockReset();
+  });
+  it('tests a file upload', async () => {
+    await request(app)
+      .post('/api/files')
+      .field('reportId', report.dataValues.id)
+      .field('attachmentType', 'ATTACHMENT')
+      .attach('File', `${__dirname}/testfiles/testfile.pdf`)
+      .expect(200)
+      .then((res) => {
+        fileId = res.body.id;
+        expect(s3Uploader).toHaveBeenCalled();
+      });
+  });
+  it('checks the metadata was uploaded to the database', async () => {
+    const file = await File.findAll({ where: { id: fileId } });
```

Comment:
This could be `File.findOne` and then all of the `file[0]` could be replaced with `file`